src/core/sync/mutex.d

```diff
@@ -122,6 +122,8 @@ class Mutex :
         if (is(Q == Mutex) || is(Q == shared Mutex))
     in
     {
+        assert(obj !is null,
+            "Error: the argument passed to Mutex.this(Object obj) is null.");
```

Comment:
Failed asserts are always an `Error`, so I would drop the "Error: " prefix and in keeping with the preexisting error message for `obj.__monitor` go with simply "The provided object must not be null.".